src/components/SearchBar.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { mobileThresholdsPixels } from './styledComponents';
+
+const FormContainer = styled.div``;
+const Input = styled.input.attrs({ type: 'text' })`
+`;
+
+const Checkbox = styled.span`
+  cursor: pointer;
+  font-size: 11px;
+  margin-right: -1px;
+  padding: 2px 7px;
+  border: 1px solid #d8d8d8;
+  border-radius: 10px 0px 0px 10px;
+  ${({ isActive }) => isActive && `
+    background-color: #3092f9ba;
+    border-color: #3092f9ba;
+  `}
+  @media(max-width: ${mobileThresholdsPixels}) {
+    border-radius: 10px;
+  }
+`;
+
+const SubmitButton = styled.button`
+  margin-left: 16px;
+`;
+
+const SearchBar = ({ handleOnBlur, handleKeyUp, isSearcAtStart, toggleIsSearcAtStart, runSearch }) => (
+  <FormContainer>
+    <Checkbox isActive={isSearcAtStart} onClick={() => toggleIsSearcAtStart()}>
+      Starts with
+    </Checkbox>
+    <Input onBlur={(e) => { handleOnBlur(e); }} onKeyUp={(e) => { handleKeyUp(e); }} />
+    <SubmitButton onClick={() => { runSearch(); }}>Search</SubmitButton>
```

Comment:
Don't create an extra anonymous function here - you can just pass in the function and it will get called appropriately: `onClick={runSearch}`. (this also applies to the `onBlur` and `onKeyUp` properties of the `Input` one line above and the `onClick` on the `Checkbox`)